### ISSUE TITLE
Connection trouble

### ISSUE THREAD

**Issue by kraemeru** [slaclab/kpix#15]
When trying to connect to KPiX either via the script or the GUI it regularly happens that no connection to the KPiX can be made. The KPiX itself is not exactly random in that some appear to have this issue more often than others.

The exact error can also vary between read attempts but is most often an error of type
DesyTrackerRoot.DesyTracker.KpixDaqCore.KpixAsicArray.KpixAsic[0].ChanModeA_2 with address 0x01100108. Error Non zero status message returned on fpga register bus in hardware: 0x2
(just an example, not the exact error)

Simply increasing the idle clock or to manually performing a "SysConfig/KpixReset" did not resolve the issue most of the time.

The issue persists only until a successful connection to the KPiX was made which most of the time is simply a matter of trying it over and over again (in gui with manually performing the resets). Afterwards further configurations resolve without any trouble until the KPiX is powered off and on again.

**Comment by kraemeru** [comment]
Forgot to mention the version that was used is
DesyTracker v1.2.3, Rogue v5.2.1

**Comment by bengineerd** [comment]
@kraemeru Are you still seeing this issue with DesyTracker v1.4.0?

**Comment by kraemeru** [comment]
@bengineerd yes and no.

I still have issues connecting to the KPiX the first time it can be basically any of the KPiX giving me an error but the error itself has shifted slightly. The error I get from basically any KPiX is

**rogue.GeneralError: Block::checkTransaction: General Error: Verify error for block DesyTrackerRoot.DesyTracker.KpixDaqCore.KpixAsicArray.KpixAsic[4].CntrlDisPerReset with address 0x011040c0. Byte: 3. Got: 0x1c, Exp: 0x18, Mask: 0xff**

but this error only appears once. Afterwards I can connect to the KPiX just fine.

But the one that repeatedly makes trouble is KPiX 9 which repeatedly gives me the same error.

**rogue.GeneralError: Block::checkTransaction: General Error: Transaction error for block DesyTrackerRoot.DesyTracker.KpixDaqCore.KpixAsicArray.KpixAsic[9].CfgAutoReadDisable with address 0x01109004. Error Non zero status message returned on fpga register bus in hardware: 0x2**


Until it at some point gives me the same error as the other KPiX can give me at the start which is

**rogue.GeneralError: Block::checkTransaction: General Error: Verify error for block DesyTrackerRoot.DesyTracker.KpixDaqCore.KpixAsicArray.KpixAsic[9].CntrlDisPerReset with address 0x011090c0. Byte: 1. Got: 0xe7, Exp: 0xc7, Mask: 0xff**

And when I then try it again it works. How many attempts it takes for me to see this though varies a lot it very much seems random to me. I got it connect once tonight but then after power cycling even after like 50 attempts I have yet to succeed in connecting it.
Once connected I can always send and receive configs etc until I power cycle the KPiX.

I added the complete output in four logs (error from random KPiX, error from KPiX 9,then random KPiX error on KpiX 9 and subsequent success and how it looks directly after a power cycle) here: https://sas.desy.de/index.php/s/ibJijCHAL4YTfEt
